android-core/plugins/org.eclipse.andmore/src/org/eclipse/andmore/internal/build/builders/BaseBuilder.java

```diff
@@ -320,10 +320,16 @@ protected void abortOnBadSetup(@NonNull IJavaProject javaProject,
         if (projectState == null) {
             projectState = Sdk.getProjectState(javaProject.getProject());
         }
-
+        
+        // the project state can still be null, 
+        // e.g. if there is no project.properties file
+        if (projectState == null) {
```

Comment:
@hrj it is a balance act as well though.   For some of these prs it may seem like unnecessary work, but ideally we want unit tests to go with outside contributions.  The reason is that along with the contribution, we are looking for people to become committers as well, and we want to make sure our code is regression tested for whoever comes along.   

If I get some time to review these again this weekend, I'll probably move the simpliest ones along after manually verifying them.